### PR TITLE
Poc multiple sld

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**

### CODE REVIEW

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/components/singleLineDiagram/utils.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                                arrayFormat,
+                            }
+                        ).toString()
+                );
+            }
+        },
+        [history, location]
+    );
+
+    return [
```
Comment (etiennehomer) [inline src/components/singleLineDiagram/utils.js:97]: We also need closeSubstationDiagram nope ?
And pinning substation doesn't work

Comment (JacquesBorsenberger) [reply src/components/singleLineDiagram/utils.js]: wrong naming

File: src/components/singleLineDiagram/utils.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        },
+        // Note: studyUuid and history don't change
+        [addToSearchParams]
+    );
+
+    const showSubstationDiagram = useCallback(
+        (substationId) => {
+            addToSearchParams(SvgType.SUBSTATION, substationId);
+        },
+        // Note: studyUuid and history don't change
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/singleLineDiagram/utils.js]: done

File: src/components/singleLineDiagram/single-line-diagram.js [hunk 3]
```diff
@@ -904,6 +956,7 @@ const SizedSingleLineDiagram = forwardRef((props, ref) => {
                 {
                     <div
                         id="sld-svg"
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/singleLineDiagram/single-line-diagram.js:957]: ref={svgRef}
is there for when we need to reference it, seach the dom for id is not very efficient

File: src/components/singleLineDiagram/single-line-diagram.js [hunk 4]
```diff
@@ -794,18 +823,18 @@ const SizedSingleLineDiagram = forwardRef((props, ref) => {
 
     const onCloseHandler = () => {
         if (props.onClose !== null) {
-            dispatch(selectItemNetwork(null));
-            dispatch(fullScreenSingleLineDiagram(false));
-            props.onClose();
+            dispatch(fullScreenSingleLineDiagram(undefined));
+            console.info(props);
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/singleLineDiagram/single-line-diagram.js]: done

File: src/components/singleLineDiagram/single-line-diagram.js [hunk 5]
```diff
@@ -304,9 +330,11 @@ const SizedSingleLineDiagram = forwardRef((props, ref) => {
             totalHeight,
             svgPreferredWidth,
             svgPreferredHeight,
-            headerPreferredHeight
+            headerPreferredHeight,
+            numberToDisplay
         );
         if (typeof sizes != 'undefined') {
+            console.info(numberToDisplay, totalWidth, sizes);
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/singleLineDiagram/single-line-diagram.js]: done

File: src/components/singleLineDiagram/single-line-diagram-pane.js [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+            parseArrays: true,
+            ignoreQueryPrefix: true,
+            arrayFormat: 'indices',
+        });
+        let newVoltageLevelIds = getArray(queryParams['views']);
+
+        setViews((oldValue) =>
+            mergeDisplayed(oldValue, newVoltageLevelIds, createView)
+        );
+        // setMinimizedSLD( (oldValue) => );
```
Comment (etiennehomer) [inline src/components/singleLineDiagram/single-line-diagram-pane.js:274]: remove

Comment (JacquesBorsenberger) [reply src/components/singleLineDiagram/single-line-diagram-pane.js]: done

**Review by FranckLecuyer — APPROVED** [review]

File: src/components/singleLineDiagram/single-line-diagram-pane.js [hunk 7]
```diff
@@ -0,0 +1,429 @@
...
+
+    const [viewState, setViewState] = useState(new Map());
+
+    const [displayedSLD, setDisplayedSld] = useState([]);
+
+    const createView = useDisplayView(network, studyUuid, workingNode);
+
+    const dispatch = useDispatch();
+
+    const [closeViwe, openVoltageLevel, openSubstation] =
```
Comment (jonenst) [inline src/components/singleLineDiagram/single-line-diagram-pane.js:207]: tyoo

**Pull request merged**